### PR TITLE
Replace table with datalist & Sort using the backend

### PR DESCRIPTION
**Pull request by jmarrero**
This adds sorting and changes the table for the patternfly datalist component. Also did some cleanup recommended by the tslinter

### CODE REVIEW

File: src/main/java/com/redhat/pantheon/data/ModuleDataRetriever.java [hunk 1]
```diff
@@ -26,6 +26,16 @@ public ModuleDataRetriever(ResourceResolver resolver) {
         this.resolver = resolver;
     }
 
+    public List<Map<String, Object>> getModulesSort(String searchTerm, String key, String direction) {
+        if (!direction.equals("desc")) {
+          direction = "asc";
+        }
+        if (!key.equals("jcr:title") && !key.equals("jcr:description") && !key.equals("jcr:created")) {
```
[the remark this answers is not in the document]
Comment (jmarrero) [reply src/main/java/com/redhat/pantheon/data/ModuleDataRetriever.java]: true, I was trying to be explicit on my intent of preventing anything else to pass thru, and prevent any SQL injection. Did not think to much on the extra check for something I am defaulting to. I'll remove it and I agree that anything more will be better on a static set.

File: src/main/java/com/redhat/pantheon/servlet/ModuleListingServlet.java [hunk 2]
```diff
@@ -33,11 +33,20 @@
     protected void doGet(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHttpServletResponse response) throws ServletException, IOException {
         ModuleDataRetriever mdr = new ModuleDataRetriever(request.getResourceResolver());
         String searchParam = "*";
+        String keyParam = "jcr:created";
+        String directionParam = "asc";
```
[the remark this answers is not in the document]
Comment (jmarrero) [reply src/main/java/com/redhat/pantheon/servlet/ModuleListingServlet.java]: sure thing.

File: src/main/java/com/redhat/pantheon/use/ModuleData.java [hunk 3]
```diff
@@ -33,4 +35,8 @@ public void init(Bindings bindings) {
     public List<Map<String, Object>> getModulesCreateSort() {
         return retriever.getModulesCreateSort(searchTerm);
     }
+    
+    public List<Map<String, Object>> getModulesSort() {
+        return retriever.getModulesSort(searchTerm, key, direction);
```
[the remark this answers is not in the document]
Comment (jmarrero) [reply src/main/java/com/redhat/pantheon/use/ModuleData.java]: I will remove this method. I agree we should not maintain the legacy stuff. If for any reason we need this we can add it later.

File: src/main/java/com/redhat/pantheon/data/ModuleDataRetriever.java [hunk 4]
```diff
@@ -26,6 +26,19 @@ public ModuleDataRetriever(ResourceResolver resolver) {
         this.resolver = resolver;
     }
 
+	public List<Map<String, Object>> getModulesSort(String searchTerm, String key, String direction) {
+		if (searchTerm == null || searchTerm.isEmpty()) {
+			searchTerm.equals("*");
+		}
+		if (key == null || (!key.equals("jcr:title") && !key.equals("jcr:description"))) {
```
Comment (benradey) [inline src/main/java/com/redhat/pantheon/data/ModuleDataRetriever.java]: Looks good! In the future, don't forget you can avoid the null check via `!"jcr:title".equals(key)` instead. :)

Comment (jmarrero) [reply src/main/java/com/redhat/pantheon/data/ModuleDataRetriever.java]: thanks for pointing that out I'll do that next time.

**Pull request merged**